### ISSUE TITLE
[Bug]: WriteGear().write causes python process to terminate

### ISSUE THREAD

**Issue by w4hns1nn** [abhiTronix/vidgear#413]
### Description

The write method causes my entire python to crash:
_Process finished with exit code -1073741819 (0xC0000005)_

I use generator function to iteratively write images to a file with WriteGear.
The exception is never reached. The process just terminates.

Details in description below.

Would be awesome if somebody could look into it.
Thank you.

### Issue Checklist

- [X] I have searched open or closed [issues](https://github.com/abhiTronix/vidgear/issues) for my problem and found nothing related or helpful.
- [X] I have read the [Documentation](https://abhitronix.github.io/vidgear/latest) and found nothing related to my problem.
- [X] I've read the [Issue Guidelines](https://abhitronix.github.io/vidgear/latest/contribution/issue/#submitting-an-issue-guidelines) and wholeheartedly agree.

### Expected behaviour

Writer writes to the file.

### Actual behaviour

The write method causes my entire python to crash:
_Process finished with exit code -1073741819 (0xC0000005)_

### Steps to reproduce

The write method causes my entire python to crash:
_Process finished with exit code -1073741819 (0xC0000005)_

I use generator function to iteratively write images to a file like:

```
output_params = {"-input_framerate": frame_rate}  # ffmpeg params
writer = WriteGear(output=save_path, compression_mode=True, **output_params)
for img in image_generator:
       try:
            if not isinstance(img, np.ndarray):
                raise ValueError("The image generator should yield images as numpy arrays or filepaths.")
            writer.write(img)
        except Exception as e:
            print(f"Error writting: {e}. Skipping frame")
```
However the writer.write(img) function causes the problem at line 430:

```
try:
   # try writing the frame bytes to the subprocess pipeline
     self.__process.stdin.write(frame.tobytes())
except (OSError, IOError):
     # log if something is wrong!
        logger.error(
          "BrokenPipeError caught, Wrong values passed to FFmpeg Pipe. Kindly Refer Docs!"
         )
        raise ValueError  # for testing purpose only
```

The exception is never reached. The process just terminates.

I found a [related problem](https://stackoverflow.com/questions/60606499/python-gets-stuck-at-pipe-stdin-writeimage-tostring) on Stack Overflow, but was not able to resolve the problem by myself.
If i use `cv2.imwrite("problem_frame.png", frame) `the image turns out normal.

Would be awesome if somebody could look into it.
Thank you.

### VidGear Version

0.3.3

### Python version

3.8

### OpenCV version

4.8

### Operating System version

Windows

**Comment by abhiTronix** [comment]
Hey @w4hns1nn, Could you please enable verbose logging by running WriteGear with `logging=True`? Once you've done that, share the full terminal output here, up to the point where you encounter the bug. Thanks!

**Comment by w4hns1nn** [comment]
Hey abhiTronix! First of all thank you very much for responding; this is honorable!
Well this morning I uninstalled ffmpg and reinstalled it; rebootet my computer and after the error disappeared.
Misterious, however I hope I can help somebody else to resolve the issue with the answer.

I'm sorry that we can't look deeper into the issue, because obviously there's deeply nested a bug :D

I closed the report. 
Thanks anyway
